go.mod

```diff
@@ -1,6 +1,7 @@
 module github.com/kubernetes-sigs/alibaba-cloud-csi-driver
 
 go 1.22.9
+toolchain go1.23.4
```

Comment:
Do we want this line?

Reply:
looks good to me for now, unless it has any sideeffects

Reply:
It will instruct local go toolchain to auto upgrade. But I think this is ignored when building container image, which leads to inconsistent version in development and production env.